### ISSUE TITLE
[Bug]: Edge Functions Beta not Working (not serving next/image images)

### ISSUE THREAD

**Issue by jzxhuang** [netlify/netlify-plugin-nextjs#1400]
### Summary

We want to use the new Edge Functions support (thanks for the work on that), but running into issues: when we enable the environment variable, images served by `next/image` fail to load. Requests to images (i.e. `/_next/image?...`) fail with `400` error. It appears that the Edge Functions aren't getting bundled/deployed correctly perhaps? 

Curiously, we see the requests getting sent to the `___netlify-handler` Netlify Function, I think they should be picked up by the Edge Function? We see the following:

```
Jun 17, 10:48:31 AM: a6ce2737 ERROR  The requested resource isn't a valid image for /_next/static/media/logo.2a28d0df.png received text/html; charset=utf-8
Jun 17, 10:48:32 AM: a6ce2737 INFO   [GET] /_next/image?q=75&url=%2F_next%2Fstatic%2Fmedia%2Flogo.2a28d0df.png&w=96 (SSR)
Jun 17, 10:48:32 AM: a6ce2737 Duration: 59.17 ms	Memory Usage: 120 MB	
```

SO it seems like the edge functions aren't handling the request? Curiously, in the `Edge Functions` tab of the Netlify UI, we see this — not sure if we should be seeing the functions themselves like in `Netlify Functions`:
<img width="700" alt="image" src="https://user-images.githubusercontent.com/26701004/174324842-62abd9af-5c18-4bd2-a8af-f51aac376a36.png">

If we disable the environment variable, images are served correctly (i.e. through the `_ipx` Netlify Function). 

**Important note:**
We use Netlify CLI to build and deploy on our own CI infrastructure. CI/CD through Netlify is disabled. We are using the latest version of the CLI as of writing (`10.5.1`) which uses `@netlify/build 27.1.5`. We run the command `netlify deploy --build` 

### Steps to reproduce

1. build plugin 4.9.1, netlify-cli 10.5.1. Set `NEXT_USE_NETLIFY_EDGE` to "true"
2.create app that renders an image with `next/image` 
3. `ntl build --deploy`
4. request to load the image results in 400 and is handled by Netlify Funciton instead of Edge Function

I don't have time to try and create an isolated repro repo atm, but I will try to create one next week if I get the time. In the meantime I'm filing this in case the Netlify team has seen this before and has ideas on how to resolve it. Thanks in advance! 

### Plugin version

4.9.1

### More information about your build

- [X] I am building using the CLI
- [ ] I am building using file-based configuration (`netlify.toml`)

### What OS are you using?

Linux

### Your netlify.toml file

<details>
  <summary>`netlify.toml`</summary>
  
```toml
# Paste content of your `netlify.toml` file here
# Some info has been redacted, I've included the key bits
[build.environment]
  NEXT_USE_NETLIFY_EDGE = "true"

[[plugins]]
  package = "@netlify/plugin-nextjs"
```

</details>

### Your public/_redirects file

<details>
  <summary>`_redirects`</summary>
  
```toml
# Paste content of your `_redirects` file here
```

</details>

### Your `next.config.js` file

<details>
  <summary>`next.config.js`</summary>
  
```toml
# Paste content of your `next.config.js` file here. Check there is no private info in there.
```

</details>

### Builds logs (or link to your logs)

<details>
  <summary>Build logs</summary>
  
```
# Paste logs here
```

</details>

### Function logs

<details>
  <summary>Function logs</summary>
  
```
# Paste logs here
Jun 17, 10:48:31 AM: a6ce2737 ERROR  The requested resource isn't a valid image for /_next/static/media/logo.2a28d0df.png received text/html; charset=utf-8
Jun 17, 10:48:32 AM: a6ce2737 INFO   [GET] /_next/image?q=75&url=%2F_next%2Fstatic%2Fmedia%2Flogo.2a28d0df.png&w=96 (SSR)
Jun 17, 10:48:32 AM: a6ce2737 Duration: 59.17 ms	Memory Usage: 120 MB	
```

</details>

### .next JSON files

<details>
  <summary>generated .next JSON files</summary>
  
```
# Paste file contents here. Please check there isn't any private info in them
# You can either build locally, or download the deploy from Netlify by clicking the arrow next to the deploy time.
```

</details>

**Comment by lourd** [comment]
@ascorbic, can you take a look at this please? This is still an issue for us even with 4.12.2. Is there something we need do to enable edge functions for our Next.js site? Do edge functions not get created when deploying with the CLI?

I see this line in the build logs

> Using Netlify Edge Functions for image format detection. Set env var "NEXT_DISABLE_EDGE_IMAGES=true" to disable.

But when looking at image requests on the deploy preview, images that were imported from source png files are still being sent as pngs instead of webp.

Deploy logs [here](https://app.netlify.com/sites/adoring-tesla-fc69b1/deploys/62d7755db37fce62376ee548)

**Comment by lourd** [comment]
Thanks @ascorbic. Just tried by upgrading the CLI to 10.12.2 and deploying. I see the logs now mention that it's bundling edge functions, but still not seeing the image optimization.

Here's the [logs from the deploy](https://drive.google.com/file/d/1CIGIE6owB0Tyr1vRz3jzJoiwuYu0Yyq9/view?usp=sharing) and [here's the image I'm testing](https://9590-fix-netlify-cli-upgrade--adoring-tesla-fc69b1.netlify.app/_ipx/w_750,q_75/%2F_next%2Fstatic%2Fmedia%2F404.ea4f2e03.png?url=%2F_next%2Fstatic%2Fmedia%2F404.ea4f2e03.png&w=750&q=75).

It says it deployed the functions from cache. Do I need to explicitly skip the cache?

**Comment by ascorbic** [comment]
Thanks. From that log it seems it's not deploying the edge function, which would explain the issue. Could you pass the `--debug` flag to the `netlify` CLI, which will print the contents of the edge functions manifest?

**Comment by lourd** [comment]
K, thanks @ascorbic. Here's [logs from a deploy with `--debug`](https://drive.google.com/file/d/1miUKhWostNU1HAc9tJ4M-vqGAWdpd4ME/view?usp=sharing)

**Comment by ascorbic** [comment]
Excellent, thanks. I've passed this over to the edge functions team, as it appears to be an issue there rather than anythign specific to Next.js. We'll get back to you ASAP

**Comment by danez** [comment]
@lourd I saw that you use `--cwd` which wasn't playing well with the edge function deployment. This case is now fixed with the latest release of `netlify-cli` version `10.11.2`. 

For the initial report in this issue, I do not see that `--cwd` was used, but there are also no build/deploy logs so hard to tell. Please try again with the latest version of `netlify-cli` and the latest version of this plugin.

If it still does not work feel free to reopen and attach the output of the cli with the `--debug` flag active.

**Comment by lourd** [comment]
Confirmed that your fix in 10.11.2 does the trick, thank you so much @danez!

**Comment by jzxhuang** [comment]
Thanks @danez and everyone for the quick fix! 🙏
